test/evergreen/configure_combinations.sh

```diff
@@ -1,59 +1,84 @@
 #!/bin/bash
 
+: ${CMAKE_BIN:=cmake}
+
+for i in "$@"; do
+  case $i in
+    -g=*|--generator=*)
+      GENERATOR="${i#*=}"
+      shift # past argument=value
+      ;;
+    -j=*|--parallel=*)
+      PARALLEL="-j ${i#*=}"
+      shift # past argument=value
+      ;;
+    *)
+      # unknown option
+      ;;
+  esac
+done
+
+if [ -z "${GENERATOR}" ]; then
+    GENERATOR="Unix Makefiles"
+fi
+if [ "$GENERATOR" != "Ninja" ] && [ "$GENERATOR" != "Unix Makefiles" ]; then
+    echo "Invalid build generator: $GENERATOR. Valid options 'Ninja', 'Unix Makefiles'"
+fi
+
+if [ "$GENERATOR" == "Unix Makefiles" ]; then
+    GENERATOR=$(echo $GENERATOR | sed -e 's/ /\\ /')
+    GENERATOR_CMD="make"
+else
+    GENERATOR_CMD="ninja"
+fi
+
 cd $(git rev-parse --show-toplevel)
 echo `pwd`
-sh build_posix/reconf
 
 curdir=`pwd`
 
-flags="CFLAGS=\"-Werror -Wall -Wextra -Waddress -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wformat-security -Wformat-nonliteral -Wformat=2 -Wmissing-declarations -Wmissing-field-initializers -Wmissing-prototypes -Wnested-externs -Wno-unused-parameter -Wpointer-arith -Wredundant-decls -Wshadow -Wundef -Wunused -Wwrite-strings -O -fno-strict-aliasing -Wuninitialized\"
-CC=clang CFLAGS=\"-Wall -Werror -Qunused-arguments -Wno-self-assign -Wno-parentheses-equality -Wno-array-bounds\""
+flags="-DCMAKE_TOOLCHAIN_FILE=$curdir/cmake/toolchains/gcc.cmake -DCMAKE_C_FLAGS=\"-Werror -Wall -Wextra -Waddress -Waggregate-return -Wbad-function-cast -Wno-missing-field-initializers -Wno-cast-function-type -Wcast-align -Wformat-security -Wformat-nonliteral -Wformat=2 -Wnested-externs -Wno-unused-parameter -Wpointer-arith -Wredundant-decls  -Wundef -Wunused -Wwrite-strings -O -fno-strict-aliasing -Wuninitialized -Wno-discarded-qualifiers -Wno-incompatible-pointer-types -Wno-int-conversion -Wno-sign-conversion -Wno-pointer-sign -Wno-unused-variable\"
+-DCMAKE_TOOLCHAIN_FILE=$curdir/cmake/toolchains/clang.cmake -DCMAKE_C_FLAGS=\"-Wall -Werror -Qunused-arguments -Wno-self-assign -Wno-parentheses-equality -Wno-array-bounds\""
 
-options="--enable-diagnostic
---disable-shared
---disable-static --enable-python
---enable-snappy --enable-zlib --enable-lz4
---with-builtins=lz4,snappy,zlib
---enable-diagnostic --enable-python
---enable-strict --disable-shared"
+options="-DHAVE_DIAGNOSTIC=1
+-DENABLE_SHARED=0 -DENABLE_STATIC=1
+-DENABLE_STATIC=0 -DENABLE_PYTHON=1
+-DENABLE_SNAPPY=1 -DENABLE_ZLIB=1 -DENABLE_LZ4=1
+-DHAVE_BUILTIN_EXTENSION_LZ4=1 -DHAVE_BUILTIN_EXTENSION_SNAPPY=1 -DHAVE_BUILTIN_EXTENSION_ZLIB=1
+-DHAVE_DIAGNOSTIC=1-DENABLE_PYTHON=1
+-DENABLE_STRICT=1 -DENABLE_STATIC=1 -DENABLE_SHARED=0 -DWITH_PIC=1"
```

Comment:
Can you remind me why `-DWITH_PIC=1` is necessary here?

Reply:
This is needed to successfully compile the `smoke` binary in  the `BuildTest` function. The `smoke` program is built as a dynamic executable, and given we we are using a static wiredtiger library, the code needs to be build with `fPIC` (otherwise we create symbol relocation errors). I alternatively tried compiling the `smoke` binary as a static executable (when using a static wiredtiger library), but this approach didn't work on our RHEL machines since it doesn't have installed static builds of `pthread`, `dl` etc. 
I felt just enabling `WITH_PIC` was the simplest approach here.